rllib/agents/trainer.py

```diff
@@ -653,50 +653,65 @@ def default_logger_creator(config):
-            elif "." in env:
+            if _global_registry.contains(ENV_CREATOR, self._env_id):
+                self.env_creator = _global_registry.get(
+                    ENV_CREATOR, self._env_id)
+
+            # A class path specifier.
+            elif "." in self._env_id:
 
                 def env_creator_from_classpath(env_context):
                     try:
-                        env_obj = from_config(env, env_context)
+                        env_obj = from_config(self._env_id, env_context)
                     except ValueError:
                         raise EnvError(
-                            ERR_MSG_INVALID_ENV_DESCRIPTOR.format(env))
+                            ERR_MSG_INVALID_ENV_DESCRIPTOR.format(
+                                self._env_id))
                     return env_obj
 
                 self.env_creator = env_creator_from_classpath
             # Try gym/PyBullet/Vizdoom.
             else:
                 self.env_creator = functools.partial(
-                    gym_env_creator, env_descriptor=env)
+                    gym_env_creator, env_descriptor=self._env_id)
+        # No env -> Env creator always returns None.
         else:
             self.env_creator = lambda env_config: None
 
-        # Merge the supplied config with the class default, but store the
-        # user-provided one.
-        self.raw_user_config = config
-        self.config = self.merge_trainer_configs(self._default_config, config,
-                                                 self._allow_unknown_configs)
-
         # Check and resolve DL framework settings.
-        # Enable eager/tracing support.
+        # Tf-eager (tf2|tfe), possibly with tracing set to True. Recommend
+        # setting tracing to True for speedups.
         if tf1 and self.config["framework"] in ["tf2", "tfe"]:
             if self.config["framework"] == "tf2" and tfv < 2:
                 raise ValueError("`framework`=tf2, but tf-version is < 2.0!")
             if not tf1.executing_eagerly():
                 tf1.enable_eager_execution()
-            logger.info("Executing eagerly, with eager_tracing={}".format(
-                self.config["eager_tracing"]))
-        if tf1 and not tf1.executing_eagerly() and \
-                self.config["framework"] != "torch":
-            logger.info("Tip: set framework=tfe or the --eager flag to enable "
-                        "TensorFlow eager execution")
+            logger.info(
+                f"Executing eagerly (framework='{self.config['framework']}'),"
+                f" with eager_tracing={self.config['eager_tracing']}. For "
+                "production workloads, make sure to set eager_tracing=True in"
+                " order to match the speed of tf-static-graph "
+                "(framework='tf')")
+        # Tf-static-graph (framework=tf): Recommend upgrading to tf2 and
+        # enabling eager tracing for similar speed.
+        elif tf1 and self.config["framework"] == "tf":
```

Comment:
haha, maybe we can say this in a more "recommendation" manner, since this user is specifically asking for tf1 here? something like:

You are using tensorflow static-graph mode. **Consider setting** framework='tf2' to upgrade to tf2.x eager execution, and eager_tracing=True to reach similar execution speed as static-graph mode.

Just a suggestion.